tests/test_logic.py

```diff
@@ -785,191 +839,87 @@ def func() -> None:
     result_data = Exception('error message')
-    submission_db_object = mocker.MagicMock()
-    submission_db_object.uuid = 'myuuid'
-    submission_db_object.filename = 'filename'
-
-    co.on_file_download_failure(result_data, current_object=submission_db_object)
-
-    co.set_status.assert_called_once_with("The file download failed. Please try again.")
-    co.file_ready.emit.assert_not_called()
-
-
-def test_Controller_on_file_downloaded_decrypt_failure(homedir, config, mocker):
-    '''
-    Using the `config` fixture to ensure the config is written to disk.
-    '''
-    mock_gui = mocker.MagicMock()
-    mock_session = mocker.MagicMock()
-    co = Controller('http://localhost', mock_gui, mock_session, homedir)
-
-    co.update_sources = mocker.MagicMock()
-    co.api_runner = mocker.MagicMock()
-    co.file_ready = mocker.MagicMock()  # signal when file is downloaded
-
-    test_filename = "1-my-file-location-msg.gpg"
-    co.api_runner.result = ("", test_filename)
-    co.set_status = mocker.MagicMock()
-
-    result_data = ('this-is-a-sha256-sum', test_filename)
-    submission_db_object = mocker.MagicMock()
-    submission_db_object.uuid = 'myuuid'
-    submission_db_object.filename = 'filename'
-    mock_gpg = mocker.patch.object(co.gpg, 'decrypt_submission_or_reply',
-                                   side_effect=CryptoError())
-    mock_storage = mocker.patch('securedrop_client.logic.storage')
-    mocker.patch('shutil.move')
-
-    co.on_file_download_success(result_data, current_object=submission_db_object)
-    mock_gpg.call_count == 1
-    co.set_status.assert_called_once_with(
-        "Failed to decrypt file, please try again or talk to your administrator.")
-    mock_storage.set_object_decryption_status_with_content.assert_called_once_with(
-        submission_db_object, mock_session, False)
-    co.file_ready.emit.assert_not_called()
 
+    co.on_file_download_failure(result_data)
 
-def test_Controller_on_file_download_user_not_signed_in(homedir, config, mocker):
-    """
-    If a user clicks the download button but is not logged in,
-    an error should appear.
-    Using the `config` fixture to ensure the config is written to disk.
-    """
-    mock_gui = mocker.MagicMock()
-    mock_session = mocker.MagicMock()
-    co = Controller('http://localhost', mock_gui, mock_session, homedir)
-    source = factory.Source()
-    file_ = db.File(source=source, uuid='uuid', size=1234, filename='1-myfile.doc.gpg',
-                    download_url='http://myserver/myfile', is_downloaded=False)
-    co.on_action_requiring_login = mocker.MagicMock()
-    co.api = None
-    submission_sdk_object = mocker.MagicMock()
-    mock_submission = mocker.patch('sdclientapi.Submission')
-    mock_submission.return_value = submission_sdk_object
-    co.on_file_download(source, file_)
-    co.on_action_requiring_login.assert_called_once_with()
-
-
-def test_Controller_on_file_download_Reply(homedir, config, mocker):
```

Comment:
You're right. I thought those callbacks were only for the downloading of submissions, but it looks like they're used for replies too. I think those need to be handled as a different case since the logic for how to handle a submission and reply download are entirely different. This seems like a case where we may have been overly aggressive in our DRYness.

Reply:
Ok, actually it looks like those callbacks were never actually used. There's no way to manually trigger a reply download, so they were wired up but never actually ran. Right?